SU2_CFD/src/solvers/CTurbSolver.cpp

```diff
@@ -92,6 +96,13 @@ void CTurbSolver::Upwind_Residual(CGeometry *geometry, CSolver **solver_containe
   const bool muscl = config->GetMUSCL_Turb();
   const bool limiter = (config->GetKind_SlopeLimit_Turb() != NO_LIMITER);
 
+  /*--- Only reconstruct flow variables if MUSCL is on for flow (requires upwind) and turbulence. ---*/
+  const bool musclFlow = config->GetMUSCL_Flow() && muscl &&
+                        (config->GetKind_ConvNumScheme_Flow() == SPACE_UPWIND);
+  /*--- Only consider flow limiters for cell-based limiters, edge-based would need to be recomputed. ---*/
+  const bool limiterFlow = (config->GetKind_SlopeLimit_Flow() != NO_LIMITER) &&
+                           (config->GetKind_SlopeLimit_Flow() != VAN_ALBADA_EDGE);
```

Comment:
Since we are introducing turbulence-related fixes in #905, I wonder if it is a good time to revisit the reconstruction logic in the turbulent solver (which currently is not adequate for centered schemes), this only affects a couple of regressions.
@jayantmukho , @economon , @clarkpede what do you think?

Reply:
Seems like you have revisited some of the logic here already? 

I haven't looked at the reconstructions before. Before these changes the MUSCL reconstruction for all (flow and turbulent) variables was only done when `MUSCL_TURB= YES` (when MUSCL reconstruction was requested for the turbulence solver as well). 

Here you have changed it such that the flow variables get reconstructed if `MUSCL_FLOW  = YES` but the turbulent variables are treated as before, i.e. they get reconstructed only when  `MUSCL_TURB= YES`. 

Are you suggesting changing the reconstruction itself? Or having an alternate reconstruction when centered schemes are used? Either way, I am for it, but don't know any better reconstruction schemes. Do you have any references I could check out? CFD wiki is best I've got so far :stuck_out_tongue:

Reply:
actually, I am surprised this didn't change more regression tests.

Reply:
Well this is all I think the logic needs. Central schemes never use MUSCL so it did not make sense to reconstruct the flow variables using gradients.
This only affects cases with central schemes + MUSCL turbulence (we seem to have two :) )

In theory what we need here is not a reconstruction, but a recomputation of the mass flux that is consistent with the main solver. I played with that once here #721 #726, by storing the mass flux from the convective residual loop. It made convergence worse since it un-staggers the solution (block Gauss-Seidel vs block Jacobi) so I did not pursue it further...

Reply:
Oh nevermind, I didn't see the `&& muscl` in the declaration of `musclFlow`. 

I am not sure why you would need a recomputation of the mass flux as opposed to flow variables. The reconstructions in the EulerSolver are identical to that in TurbSolver (except for things like the low mach corrections and non-physical points). I can see an argument made for copying all those corrections in the TurbSolver as well.

Reply:
If you want the mass fluxes to respect the discretized continuity equation then they should be consistent with what is computed by the flow solver. In any case this did not seem to make a huge difference... But it is the recommended approach especially for incompressible methods.